examples/linear_model/plot_ols.py

```diff
@@ -56,25 +58,36 @@
 # Plotting the results
 # --------------------
 #
-# Finally, we visualize the results on the test set.
+# Finally, we visualize the results on the train and test data.
 import matplotlib.pyplot as plt
 
-fig, ax = plt.subplots()
+fig, ax = plt.subplots(ncols=2, figsize=(10, 5), sharex=True, sharey=True)
 
-ax.scatter(X_test, y_test, label="Data points")
-ax.plot(X_test, y_pred, linewidth=3, color="tab:orange", label="Model predictions")
-ax.set(xlabel="Feature", ylabel="Target", title="Linear Regression")
-ax.legend()
+ax[0].scatter(X_train, y_train, label="Train data points")
+ax[0].plot(X_train, regressor.predict(X_train), linewidth=3, color="tab:orange", label="Model predictions")
+ax[0].set(xlabel="Feature", ylabel="Target", title="Train set")
+ax[0].legend()
+
+ax[1].scatter(X_test, y_test, label="Test data points")
+ax[1].plot(X_test, y_pred, linewidth=3, color="tab:orange", label="Model predictions")
+ax[1].set(xlabel="Feature", ylabel="Target", title="Test set")
+ax[1].legend()
+
+fig.suptitle("Linear Regression")
 
 plt.show()
 
 # %%
 # Conclusion
 # ----------
 #
-# This example shows how to use the simplest linear model called
-# :class:`~sklearn.linear_model.LinearRegression` in scikit-learn. For this purpose, we
-# use a single feature from the diabetes dataset and try to predict the diabetes
-# progression using this linear model. We therefore load the diabetes dataset and split
-# it into training and test sets.
+# The trained model corresponds to the estimator that minimizes the mean squared error
+# between the predicted and the true target values on the training data. This means that
+# there is no other model that fits the training data better in terms of squared error.
```

Comment:
There is no other model of linear form with 2 parameters (or 3, if we have an intercept).

Reply:
I don't mean a different model in terms of parametrization but rather the same parametric model minimizing another loss for instance. Not sure how to formulate this.

Reply:
Just removing the statement.